src/Symfony/Component/HttpFoundation/UriSigner.php

```diff
@@ -11,34 +11,47 @@
 
 namespace Symfony\Component\HttpFoundation;
 
+use Symfony\Component\HttpFoundation\Exception\LogicException;
+
 /**
  * @author Fabien Potencier <fabien@symfony.com>
  */
 class UriSigner
 {
     private string $secret;
-    private string $parameter;
+    private string $hashParameter;
+    private string $expirationParameter;
 
     /**
-     * @param string $parameter Query string parameter to use
+     * @param string $hashParameter       Query string parameter to use
+     * @param string $expirationParameter Query string parameter to use for expiration
      */
-    public function __construct(#[\SensitiveParameter] string $secret, string $parameter = '_hash')
+    public function __construct(#[\SensitiveParameter] string $secret, string $hashParameter = '_hash', string $expirationParameter = '_expiration')
     {
         if (!$secret) {
             throw new \InvalidArgumentException('A non-empty secret is required.');
         }
 
         $this->secret = $secret;
-        $this->parameter = $parameter;
+        $this->hashParameter = $hashParameter;
+        $this->expirationParameter = $expirationParameter;
     }
 
     /**
      * Signs a URI.
      *
      * The given URI is signed by adding the query string parameter
      * which value depends on the URI and the secret.
+     *
+     * @param \DateTimeInterface|\DateInterval|int|null $expiration The expiration for the given URI.
+     *                                                              If $expiration is a \DateTimeInterface, it's expected to be the exact date + time.
+     *                                                              If $expiration is a \DateInterval, the interval is added to "now" to get the date + time.
+     *                                                              If $expiration is an int, it's expected to be a timestamp in seconds of the exact date + time.
+     *                                                              If $expiration is null, no expiration.
+     *
+     * The expiration is added as a query string parameter.
      */
-    public function sign(string $uri): string
+    public function sign(string $uri, \DateTimeInterface|\DateInterval|int|null $expiration = null): string
```

Comment:
I missed that this is a BC break. Instead, we should comment the argument on the signature and read it with func_get_args (see on 6.4 for similar new arguments).

Reply:
Is it a BC even if the default value does not change the old behavior of the method ?
Do you want another PR linked to a new issue to fix this ?

Reply:
It's a BC break for classes that extend this class. Yes, it'd be great to fix it, PR welcome :)